File: app/src/main/java/fr/xgouchet/chronorg/ui/viewholders/BaseViewHolder.java
package fr.xgouchet.chronorg.ui.viewholders;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * @author Xavier Gouchet
 */
public abstract class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    protected BaseViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bindItem(@NonNull T item);
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/viewholders/BaseViewHolder.kt
package fr.xgouchet.khronorg.ui.viewholders

import android.support.v7.widget.RecyclerView
import android.view.View
import io.reactivex.functions.Consumer

/**
 * @author Xavier Gouchet
 */
abstract class BaseViewHolder<T>(val listener: Consumer<T>?, itemView: View) : RecyclerView.ViewHolder(itemView) {

    var item: T? = null


    fun bindItem(item: T) {
        this.item = item
        onBindItem(item)
    }

    protected fun fireSelected() {
        item?.let {
            listener?.accept(it)
        }
    }

    protected abstract fun onBindItem(item: T)
}
